src/OrleansRuntime/GrainDirectory/LocalGrainDirectory.cs

```diff
@@ -1088,6 +1098,11 @@ internal IRemoteGrainDirectory GetDirectoryReference(SiloAddress silo)
             return InsideRuntimeClient.Current.InternalGrainFactory.GetSystemTarget<IRemoteGrainDirectory>(Constants.DirectoryServiceId, silo);
         }
 
+        private bool IsSiloNextInTheRing(SiloAddress siloAddr, int hash, bool excludeMySelf)
```

Comment:
This logic is same as other logic, except with <= rather than >=.
Please consider using one static function with both hashes in arguments, with the hash order changed, rather then two functions with the comparison reversed.